documentation/proposals/Proposal - Generic Math.md

```diff
@@ -0,0 +1,1189 @@
+    public static Vector2I<TScalar> operator -(Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> operator *(Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> operator /(Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> operator %(Vector2I<TScalar> left, Vector2I<TScalar> right);
+
+    public static Vector2I<TScalar> operator *(Vector2I<TScalar> left, TScalar right);
+    public static Vector2I<TScalar> operator /(Vector2I<TScalar> left, TScalar right);
+    public static Vector2I<TScalar> operator %(Vector2I<TScalar> left, TScalar right);
+
+    public static Vector2I<TScalar> operator *(TScalar left, Vector2I<TScalar> right);
+
+    public static Vector2I<TScalar> operator +(Vector2I<TScalar> value);
+    public static Vector2I<TScalar> operator -(Vector2I<TScalar> value);
+
+    public static Vector2I<TScalar> operator &(Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> operator |(Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> operator ^(Vector2I<TScalar> left, Vector2I<TScalar> right);
+
+    public static Vector2I<TScalar> operator ~(Vector2I<TScalar> value);
+
+    public TScalar LengthSquared { get; }
+
+    public Vector2I<TScalar> Dot(Vector2I<TScalar> other);
+    public TScalar DistanceSquared(Vector2I<TScalar> other);
+    public Vector2I<TScalar> Abs();
+    public void CopyTo(TScalar[] array);
+    public void CopyTo(TScalar[] array, int index);
+
+    public static Vector2I<TScalar> Dot (Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> DistanceSquared (Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> Clamp (Vector2I<TScalar> value, Vector2I<TScalar> min, Vector2I<TScalar> max);
+    public static Vector2I<TScalar> Min (Vector2I<TScalar> value1, Vector2I<TScalar> value2);
+    public static Vector2I<TScalar> Max (Vector2I<TScalar> value1, Vector2I<TScalar> value2);
+    public static Vector2I<TScalar> Abs (Vector2I<TScalar> value);
+}
+```
+
+```cs
+public readonly record struct Vector2F<TScalar> : IEquatable<Vector2F<TScalar>>, 
+                           IEquatable<Vector2>,
+                           IFloatingPointVector<Vector2F<TScalar>, TScalar>,
+                           IVector2<Vector2F<TScalar>, TScalar>
+                           where TScalar : IFloatingPointIeee754<TScalar>
+{
+    ///<summary>
+    ///The X component of the Vector
+    ///</summary>
+    [DataMember]
+    public TScalar X { get; }
+
+    ///<summary>
+    ///The Y component of the Vector
+    ///</summary>
+    [DataMember]
+    public TScalar Y { get; }
+
+    public static Vector2F<TScalar> UnitX { get; }
+    public static Vector2F<TScalar> UnitY { get; }
+
+    public static Vector2F<TScalar> Zero { get; }
+    public static Vector2F<TScalar> One { get; }
```

Comment:
We've approved and are adding `PositiveInfinity`, `NegativeInfinity`, `NaN`, `Epsilon`, `NegativeZero`, `Pi`, `Tau`, and `E` in System.Numerics.

Reply:
Is this just setting every component to these values?

Reply:
Yep!

Notably `AllBitsSet` is also a common one

Reply:
Adding